feat: New time range label

kgabryje opened the pull request:
### SUMMARY
This PR implements the new design for time range pill. Functionality wise it works the same as before, except that I added additional tooltip when the label is truncated.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->


https://user-images.githubusercontent.com/15073128/205328748-10f7a70b-1caf-4ecb-9df7-ba75023c82ab.mov



### TESTING INSTRUCTIONS
Test the new time range pill in native filters (horizontal and vertical), in native filters config modal and in Explore adhoc filters.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

kgabryje committed:
feat: Implement new time range label

kgabryje committed:
Add tooltip when label is truncated

zhaoyongjie commented on superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:
```diff
@@ -45,6 +45,7 @@ import {
   CustomFrame,
   AdvancedFrame,
 } from './components';
+import { DateLabel } from './components/DateLabel';
```
kindly move the `DateLabel` into `index.ts` in the `components` so that import like above Frame. for instance:

```typescript
import {
  CommonFrame,
  CalendarFrame,
  CustomFrame,
  AdvancedFrame,
  DateLabel
} from './components';
```

kgabryje replied:
Good point, done!

kgabryje committed:
Fix import

michael-s-molina commented on superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:
```diff
@@ -172,16 +174,27 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
+          setTooltipTitle(labelIsTruncated ? value : null);
         } else {
+          const tooltipTitle = labelIsTruncated ? (
+            <>
```
Is this fragment necessary?

kgabryje replied:
Nope! Removed

michael-s-molina commented on superset-frontend/src/explore/components/controls/DateFilterControl/components/DateLabel.tsx:
```diff
@@ -0,0 +1,97 @@
+};
+
+// This is the color that antd components (such as Select or Input) use on hover
+// TODO: use theme.colors.primary.base here and in antd components
```
Can we just set it as the primary color already?

michael-s-molina replied:
Makes sense. Thanks for the context!

kgabryje committed:
Remove unnecessary Fragment

michael-s-molina commented:
There's no red border indicating a validation error.

https://user-images.githubusercontent.com/70410625/205342086-fbe94188-600f-424e-bd42-b37598f377b5.mov

villebro commented:
I love the new design! ❤️  I will look more closely later today/the weekend if this isn't merged before then.

kgabryje committed:
Add error border

kgabryje commented:
Well spotted @michael-s-molina! Fixed + I cleaned up some old css

<img width="441" alt="image" src="https://user-images.githubusercontent.com/15073128/205361198-7fbdd3f4-f339-4a6a-a2eb-c884a5f86a5e.png">

kgabryje commented:
@eric-briscoe could you review as well please?

codyml reviewed (approved):
LGTM!  I opened a [PR](https://github.com/apache/superset/pull/22324) to help with a tooltip bug caused by the `useCSSTextTruncation` hook that shows up in this PR, but it's not related to this PR's code so no need to include here.  I also noticed the same thing as in a previous PR about popovers for filters in the overflow dropdown, where clicking Cancel or Apply closes the overflow dropdown preemptively, but we already talked about fixing that in a separate PR.

zhaoyongjie commented on superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:
```diff
@@ -139,21 +140,22 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
 
   useEffect(() => {
     if (value === NO_TIME_RANGE) {
       setActualTimeRange(NO_TIME_RANGE);
-      setTooltipTitle(NO_TIME_RANGE);
+      setTooltipTitle(null);
```
`NO_TIME_RANGE` represents a time filter constant, which means `No filter` in the time filter, so I think the original tooltip title should be kept.

kgabryje replied:
We want tooltip to display the label text if label is truncated + the actual time range of it's different than human readable label.
In the case of no filter both the label and the tooltip said "No filter" which was redundant.

zhaoyongjie commented on superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx:
```diff
@@ -172,16 +174,25 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
           setActualTimeRange(value);
+          setTooltipTitle(labelIsTruncated ? value : null);
         } else {
+          const tooltipTitle = labelIsTruncated ? (
```
<img width="457" alt="image" src="https://user-images.githubusercontent.com/2016594/205433586-02af27a5-7039-41c3-88ed-e4ffe3080d1e.png">

Here, the tooltip title is either HRT(human-readable text) **or** ADR(actual date time range). I know the design goal is when the value is truncated, the untruncated value should be displayed in the tooltip title. In practice, the effect is weird, --- there are multiple lines but I don't know which one is HRT, and which one is ADR.

BTW, the width of the tooltips seems too narrow.

kgabryje replied:
Yep that wasn't the intention! Thanks for spotting that

zhaoyongjie reviewed (commented):
Leave some comments about the style and design. The new Date Label is better than before. Thanks for the improvement!

kgabryje committed:
Add spacing in tooltip

kgabryje commented:
/testenv up FEATURE_HORIZONTAL_FILTER_BAR=true

kgabryje commented:
@villebro Thanks for review!
I don't think we display a tooltip for "Last year" in the current date pill either. Moreover, there's a bug which causes the tooltip related to previous selection to appear.


https://user-images.githubusercontent.com/15073128/205664813-f56beea2-e3f5-471f-9b55-f5c78fb6095e.mov

As for the "double" tooltip... that looks like html title. I'll try to remove it

villebro commented:
> I don't think we display a tooltip for "Last year" in the current date pill either

My comments may well be relevant for the current functionality, too. 

> Moreover, there's a bug which causes the tooltip related to previous selection to appear.

Was this already there in the previous version?

> As for the "double" tooltip... that looks like html title. I'll try to remove it

👍

kgabryje committed:
Added actual range tooltip to Last/Previous frames

kgabryje commented:
@villebro Added tooltip with actual range for Last/Previous frames 
<img width="320" alt="image" src="https://user-images.githubusercontent.com/15073128/205674321-2b07b893-1bf2-40be-a48b-11e1ee8338ee.png">

Unfortunately the double tooltip is a Safari "feature" - Safari adds it automatically for truncated texts :(

villebro reviewed (approved):
LGTM!
